procrastinate/aiopg_connector.py

```diff
@@ -125,60 +44,100 @@ def __init__(
+        kwargs.setdefault("enable_json", False)
+        kwargs.setdefault("enable_hstore", False)
+        kwargs.setdefault("enable_uuid", False)
+        conn = await aiopg.connect(**kwargs)
+        if self.json_loads:
+            psycopg2.extras.register_default_jsonb(conn.raw, loads=self.json_loads)
+        self._connection = conn
+        return conn
+
     async def execute_query(self, query: str, **arguments: Any) -> None:
-        await execute_query(
-            await self.get_connection(),
-            query=query,
-            json_dumps=self.json_dumps,
-            **arguments
-        )
+        connection = await self._get_connection()
+        # aiopg can work with psycopg2's cursor class
+        async with connection.cursor(cursor_factory=RealDictCursor) as cursor:
+            await cursor.execute(query, self._wrap_json(arguments))
 
     async def execute_query_one(self, query: str, **arguments: Any) -> Dict[str, Any]:
-        return await execute_query_one(
-            await self.get_connection(),
-            query=query,
-            json_dumps=self.json_dumps,
-            **arguments
-        )
+        connection = await self._get_connection()
+        # aiopg can work with psycopg2's cursor class
+        async with connection.cursor(cursor_factory=RealDictCursor) as cursor:
+            await cursor.execute(query, self._wrap_json(arguments))
+
+            return await cursor.fetchone()
 
     async def execute_query_all(
         self, query: str, **arguments: Any
     ) -> List[Dict[str, Any]]:
-        return await execute_query_all(
-            await self.get_connection(),
-            query=query,
-            json_dumps=self.json_dumps,
-            **arguments
-        )
+        connection = await self._get_connection()
+        # aiopg can work with psycopg2's cursor class
+        async with connection.cursor(cursor_factory=RealDictCursor) as cursor:
+            await cursor.execute(query, self._wrap_json(arguments))
+
+            return await cursor.fetchall()
 
     def make_dynamic_query(self, query: str, **identifiers: str) -> str:
-        return make_dynamic_query(query=query, **identifiers)
+        return psycopg2.sql.SQL(query).format(
+            **{
+                key: psycopg2.sql.Identifier(value)
+                for key, value in identifiers.items()
+            }
+        )
+
+    async def wait_for_activity(self):
+        connection = await self._get_connection()
+        try:
+            await asyncio.wait_for(
+                connection.notifies.get(), timeout=self.socket_timeout
+            )
+        except asyncio.TimeoutError:
+            pass
 
-    async def wait_for_jobs(self):
-        return await wait_for_jobs(
-            connection=await self.get_connection(), socket_timeout=self.socket_timeout
+    # This can be called from a signal handler, better not do async stuff
+    def interrupt_wait(self):
+        if not self._connection:
```

Comment:
I think we need to have `if not self._connection or self._connection.closed` here.